crates/primitives/src/compression/mod.rs

```diff
@@ -1,16 +1,26 @@
+#[cfg(feature = "std")]
 use std::{cell::RefCell, thread_local};
+#[cfg(not(feature = "std"))]
+use zstd::bulk::Decompressor;
+#[cfg(feature = "std")]
 use zstd::bulk::{Compressor, Decompressor};
 
 #[cfg(not(feature = "std"))]
 use alloc::vec::Vec;
 
+#[cfg(not(feature = "std"))]
+use alloc::string::ToString;
+
+#[cfg(feature = "std")]
 /// Compression/Decompression dictionary for `Receipt`.
 pub static RECEIPT_DICTIONARY: &[u8] = include_bytes!("./receipt_dictionary.bin");
+#[cfg(feature = "std")]
```

Comment:
There is a lot of feature gating here. Instead of `use` on top of the file could we just use the full path. 
For example instead of:
```
#[cfg(not(feature = "std"))]
 use alloc::vec::Vec;
```
we could: `alloc::vec::Vec::new()`

?